--- a/cosmic-comp/Cargo.toml
+++ b/cosmic-comp/Cargo.toml
@@ -73,7 +73,7 @@ features = [
   "backend_libinput",
   "backend_session_libseat",
   "backend_udev",
-  "backend_winit",
+  #"backend_winit",
   "backend_vulkan",
   "backend_x11",
   "desktop",
@@ -95,7 +95,7 @@ rev = "cdc652e0"
 
 [features]
 debug = ["egui", "egui_plot", "smithay-egui", "anyhow/backtrace"]
-default = ["systemd"]
+default = []
 systemd = ["libsystemd"]
 profile-with-tracy = ["profiling/profile-with-tracy"]
 
