--- a/src/lib.rs
+++ b/src/lib.rs
@@ -15,7 +15,7 @@
 
 use std::{error::Error, fmt::Display};
 
-#[cfg(any(target_os = "android", target_arch = "wasm32"))]
+#[cfg(any(target_arch = "wasm32"))]
 compile_error!("You are compiling for an unimplemented platform!\nContributions are welcome to current_locale to implement any new platforms.");
 
 #[cfg(target_os = "windows")]
@@ -28,7 +28,7 @@
 
 #[cfg(all(
     unix,
-    not(any(target_os = "macos", target_os = "ios", target_os = "android"))
+    not(any(target_os = "macos", target_os = "ios"))
 ))]
 #[path = "unix.rs"]
 mod imp;
--- a/src/unix.rs
+++ b/src/unix.rs
@@ -1,6 +1,6 @@
 #![cfg(all(
     unix,
-    not(any(target_os = "android", target_os = "ios", target_os = "macos"))
+    not(any(target_os = "ios", target_os = "macos"))
 ))]
 
 use crate::{ErrorKind, LocaleError};
