--- a/build.rs
+++ b/build.rs
@@ -89,6 +89,8 @@
         if target.contains("linux") && (target.contains("-gnu") || target.contains("-musl")) {
             // intl is part of glibc and musl
             return;
+        } else if target.contains("android") {
+            return;
         } else if target.contains("windows") && target.contains("-gnu") {
             // gettext doesn't come with a pkg-config file
             let gnu_root = get_windows_gnu_root();
