diff --git a/discover/gpu_linux.go b/discover/gpu_linux.go
index a2c1d87..3a2abf2 100644
--- a/discover/gpu_linux.go
+++ b/discover/gpu_linux.go
@@ -57,15 +57,12 @@ var (
 )
 
 var VulkanGlobs = []string{
-	"/usr/lib/x86_64-linux-gnu/libvulkan.so*",
-	"/usr/lib/aarch64-linux-gnu/libvulkan.so*",
-	"/usr/lib*/libvulkan.so*",
+	"@TERMUX_PREFIX@/lib/libvulkan.so*",
 }
 
 var capLinuxGlobs = []string{
-	"/usr/lib/x86_64-linux-gnu/libcap.so*",
-	"/usr/lib/aarch64-linux-gnu/libvulkan.so*",
-	"/usr/lib*/libcap.so*",
+	"@TERMUX_PREFIX@/lib/libcap.so*",
+	"@TERMUX_PREFIX@/lib/libvulkan.so*",
 }
 
 func FindLibCapLibs() []string {
