--- a/pkg/sshutil/sshutil.go
+++ b/pkg/sshutil/sshutil.go
@@ -486,7 +486,7 @@ func detectValidPublicKey(content string) bool {
 
 func detectAESAcceleration() bool {
 	if !cpu.Initialized {
-		if runtime.GOOS == "linux" && runtime.GOARCH == "arm64" {
+		if (runtime.GOOS == "linux" || runtime.GOOS == "android") && runtime.GOARCH == "arm64" {
 			// cpu.Initialized seems to always be false, even when the cpu.ARM64 struct is filled out
 			// it is only being set by readARM64Registers, but not by readHWCAP or readLinuxProcCPUInfo
 			return cpu.ARM64.HasAES
