--- a/include/bsd/sys/endian.h
+++ b/include/bsd/sys/endian.h
@@ -28,12 +28,12 @@
 
 #ifdef LIBBSD_OVERLAY
 #include <sys/cdefs.h>
-#if __has_include_next(<endian.h>)
+#if __has_include_next(<endian.h>) && !defined(__ANDROID__)
 #include_next <endian.h>
 #endif
 #else
 #include <bsd/sys/cdefs.h>
-#if __has_include(<endian.h>)
+#if __has_include(<endian.h>) && !defined(__ANDROID__)
 #include <endian.h>
 #endif
 #endif
