diff -uNr gl4es/src/CMakeLists.txt gl4es.mod/src/CMakeLists.txt
--- gl4es/src/CMakeLists.txt	2023-01-17 10:13:23.687408484 +0800
+++ gl4es.mod/src/CMakeLists.txt	2023-01-17 10:16:01.067408372 +0800
@@ -225,10 +225,10 @@
     endif()
     install(TARGETS GL
       LIBRARY
-      DESTINATION "/usr/lib/gl4es/"
+      DESTINATION "@TERMUX_PREFIX@/lib/gl4es/"
     )
     install(FILES "../include/gl4esinit.h" "../include/gl4eshint.h"
-      DESTINATION "/usr/include/gl4es/"
+      DESTINATION "@TERMUX_PREFIX@/include/gl4es/"
     )
 endif()
 
