--- a/coregrind/pub_core_transtab.h
+++ b/coregrind/pub_core_transtab.h
@@ -161,7 +161,7 @@
 #if defined(VGPV_arm_linux_android) \
     || defined(VGPV_x86_linux_android) \
     || defined(VGPV_mips32_linux_android) \
-    || defined(VGPV_arm64_linux_android)
+    || defined(VGPV_arm64_linux_android) || defined(VGPV_amd64_linux_android)
 # define N_SECTORS_DEFAULT 12
 #else
 # define N_SECTORS_DEFAULT 32
