--- a/configure.ac
+++ b/configure.ac
@@ -3764,20 +3764,15 @@
     enable_qt="no"
   ])
 ])
-have_qt5_x11="no"
+have_qt5_x11="yes"
 AS_IF([test "${enable_qt}" != "no"], [
-  PKG_CHECK_MODULES([QT], [Qt5Core >= 5.5.0 Qt5Widgets Qt5Gui Qt5Svg], [
-      PKG_CHECK_MODULES([QT5_X11], [Qt5X11Extras], [
-          have_qt5_x11="yes"
-      ],[
-          AC_MSG_WARN([Not building Qt Interface with X11 helpers.])
-      ])
-      QT_PATH="$(eval $PKG_CONFIG --variable=exec_prefix Qt5Core)"
-      QT_HOST_PATH="$(eval $PKG_CONFIG --variable=host_bins Qt5Core)"
-      QT_VERSION="$(eval $PKG_CONFIG --modversion Qt5Gui)"
-      AC_PATH_PROGS(MOC, [moc-qt5 moc], moc, ["${QT_HOST_PATH}" "${QT_PATH}/bin"])
-      AC_PATH_PROGS(RCC, [rcc-qt5 rcc], rcc, ["${QT_HOST_PATH}" "${QT_PATH}/bin"])
-      AC_PATH_PROGS(UIC, [uic-qt5 uic], uic, ["${QT_HOST_PATH}" "${QT_PATH}/bin"])
+  PKG_CHECK_MODULES([QT], [Qt6Core5Compat >= 5.5.0 Qt6Widgets Qt6Gui Qt6Svg], [
+      QT_PATH="$(eval $PKG_CONFIG --variable=exec_prefix Qt6Core5Compat)"
+      QT_HOST_PATH="$(eval qmake6 -query QT_HOST_LIBEXECS)"
+      QT_VERSION="$(eval $PKG_CONFIG --modversion Qt6Gui)"
+      AC_PATH_PROGS(MOC, [moc-qt6 moc], moc, ["${QT_HOST_PATH}" "${QT_PATH}/bin"])
+      AC_PATH_PROGS(RCC, [rcc-qt6 rcc], rcc, ["${QT_HOST_PATH}" "${QT_PATH}/bin"])
+      AC_PATH_PROGS(UIC, [uic-qt6 uic], uic, ["${QT_HOST_PATH}" "${QT_PATH}/bin"])
     ], [
       AS_IF([test -n "${enable_qt}"],[
         AC_MSG_ERROR([${QT_PKG_ERRORS}.])
