diff -uNr a/meson.build b/meson.build
--- a/meson.build	2024-07-26 17:43:32.000000000 +0800
+++ b/meson.build	2024-09-19 16:42:45.116017459 +0800
@@ -186,6 +186,8 @@
 
     ui_cargo_options = ['--manifest-path', cargo_manifest_path]
     ui_cargo_options += ['--target-dir', cargo_target_dir]
+    ui_cargo_options += ['--target', '@CARGO_TARGET_NAME@']
+    ui_cargo_options += ['--config', '"target.@CARGO_TARGET_NAME@.linker = @CC@"']
     ui_cargo_options += ['-p', 'rnote']
 
     if profile == 'default'
