--- a/meson.build
+++ b/meson.build
@@ -1,4 +1,4 @@
-project('rnote', ['rust', 'cpp'], version: '0.12.0', meson_version: '>= 1.0')
+project('rnote', ['cpp'], version: '0.12.0', meson_version: '>= 1.0')
 # add a patch suffix for alpha or beta versions in format '-<alpha|beta>.<x>'.
 patch = ''
 
@@ -63,7 +63,6 @@
 dependency('poppler', version: '>= 20.09')
 # UI/CLI specific dependencies are declared below
 
-cargo = find_program('cargo', required: true)
 # cmake is needed by the "ink-stroke-modeler-rs" bindings crate to build the C++ library
 find_program('cmake', required: true)
 find_program('glib-compile-resources', required: true)
@@ -90,7 +89,7 @@
 subdir('crates/rnote-compose')
 subdir('crates/rnote-engine')
 
-
+if false
 # cargo fmt check
 run_target(
     'cargo-fmt-check',
@@ -133,6 +132,7 @@
     ],
     env: cargo_env,
 )
+endif
 
 # build/install the ui
 if build_ui == true
@@ -174,6 +174,7 @@
         ui_output = app_name
     endif
 
+if false
     # ui cargo check
     run_target(
         'ui-cargo-check',
@@ -215,6 +216,7 @@
             meson.project_build_root() / '@OUTPUT@',
         ],
     )
+endif
 
     meson.add_install_script(
         'build-aux/meson_post_install.py',
@@ -227,7 +229,7 @@
         glib_compile_schemas: true,
         gtk_update_icon_cache: true,
         update_desktop_database: true,
-        update_mime_database: true,
+        update_mime_database: false,
     )
 
     # Windows installer
