diff -uNr ninja-1.12.0/configure.py ninja-1.11.1/configure.py
--- ninja-1.12.0/configure.py	2024-04-12 00:44:05.000000000 +0800
+++ ninja-1.11.1/configure.py	2022-08-31 03:47:02.000000000 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/env python
 #
 # Copyright 2001 Google Inc. All Rights Reserved.
 #
@@ -19,9 +19,12 @@
 Projects that use ninja themselves should either write a similar script
 or use a meta-build system that supports Ninja output."""
 
+from __future__ import print_function
+
 from optparse import OptionParser
 import os
-import shlex
+import pipes
+import string
 import subprocess
 import sys
 
@@ -261,7 +264,7 @@
 env_keys = set(['CXX', 'AR', 'CFLAGS', 'CXXFLAGS', 'LDFLAGS'])
 configure_env = dict((k, os.environ[k]) for k in os.environ if k in env_keys)
 if configure_env:
-    config_str = ' '.join([k + '=' + shlex.quote(configure_env[k])
+    config_str = ' '.join([k + '=' + pipes.quote(configure_env[k])
                            for k in configure_env])
     n.variable('configure_env', config_str + '$ ')
 n.newline()
