diff -uNr pypy2.7-v7.3.6-src/rpython/translator/platform/linux.py pypy2.7-v7.3.6-src.mod/rpython/translator/platform/linux.py
--- pypy2.7-v7.3.6-src/rpython/translator/platform/linux.py	2021-10-03 14:36:11.000000000 +0800
+++ pypy2.7-v7.3.6-src.mod/rpython/translator/platform/linux.py	2021-10-25 13:10:33.075390678 +0800
@@ -11,7 +11,7 @@
     link_flags = tuple(
                  ['-pthread',]
                  + os.environ.get('LDFLAGS', '').split())
-    extra_libs = ('-lrt',)
+    #extra_libs = ('-lrt',)
     cflags = tuple(
              ['-O3', '-pthread', '-fomit-frame-pointer',
               '-Wall', '-Wno-unused', '-Wno-address']
@@ -29,12 +29,12 @@
 
     def _include_dirs_for_libffi(self):
         return self._pkg_config("libffi", "--cflags-only-I",
-                                ['/usr/include/libffi'],
+                                ['@TERMUX_PREFIX@/include/libffi'],
                                 check_result_dir=True)
 
     def _library_dirs_for_libffi(self):
         return self._pkg_config("libffi", "--libs-only-L",
-                                ['/usr/lib/libffi'],
+                                ['@TERMUX_PREFIX@/lib/libffi'],
                                 check_result_dir=True)
 
