--- a/compiler/rustc_target/src/spec/targets/aarch64_linux_android.rs
+++ b/compiler/rustc_target/src/spec/targets/aarch64_linux_android.rs
@@ -1,11 +1,15 @@
 use crate::spec::{
-    FramePointer, SanitizerSet, StackProbeType, Target, TargetMetadata, TargetOptions, base,
+    Cc, FramePointer, LinkerFlavor, Lld, SanitizerSet, StackProbeType, Target, TargetMetadata,
+    TargetOptions, base,
 };
 
 // See https://developer.android.com/ndk/guides/abis.html#arm64-v8a
 // for target ABI requirements.
 
 pub(crate) fn target() -> Target {
+    let mut base = base::android::opts();
+    base.add_pre_link_args(LinkerFlavor::Gnu(Cc::Yes, Lld::No), &["--rtlib=compiler-rt"]);
+
     Target {
         llvm_target: "aarch64-linux-android".into(),
         metadata: TargetMetadata {
@@ -33,7 +37,7 @@
                 | SanitizerSet::SHADOWCALLSTACK
                 | SanitizerSet::ADDRESS,
             supports_xray: true,
-            ..base::android::opts()
+            ..base
         },
     }
 }
--- a/compiler/rustc_target/src/spec/targets/arm_linux_androideabi.rs
+++ b/compiler/rustc_target/src/spec/targets/arm_linux_androideabi.rs
@@ -1,6 +1,11 @@
-use crate::spec::{FloatAbi, SanitizerSet, Target, TargetMetadata, TargetOptions, base};
+use crate::spec::{
+    Cc, FloatAbi, LinkerFlavor, Lld, SanitizerSet, Target, TargetMetadata, TargetOptions, base
+};
 
 pub(crate) fn target() -> Target {
+    let mut base = base::android::opts();
+    base.add_pre_link_args(LinkerFlavor::Gnu(Cc::Yes, Lld::No), &["--rtlib=compiler-rt"]);
+
     Target {
         llvm_target: "arm-linux-androideabi".into(),
         metadata: TargetMetadata {
@@ -19,7 +24,7 @@
             features: "+strict-align,+v5te".into(),
             supported_sanitizers: SanitizerSet::ADDRESS,
             max_atomic_width: Some(32),
-            ..base::android::opts()
+            ..base
         },
     }
 }
--- a/compiler/rustc_target/src/spec/targets/armv7_linux_androideabi.rs
+++ b/compiler/rustc_target/src/spec/targets/armv7_linux_androideabi.rs
@@ -13,6 +13,8 @@
 pub(crate) fn target() -> Target {
     let mut base = base::android::opts();
     base.add_pre_link_args(LinkerFlavor::Gnu(Cc::Yes, Lld::No), &["-march=armv7-a"]);
+    base.add_pre_link_args(LinkerFlavor::Gnu(Cc::Yes, Lld::No), &["--rtlib=compiler-rt"]);
+
     Target {
         llvm_target: "armv7-none-linux-android".into(),
         metadata: TargetMetadata {
--- a/compiler/rustc_target/src/spec/targets/i686_linux_android.rs
+++ b/compiler/rustc_target/src/spec/targets/i686_linux_android.rs
@@ -1,5 +1,6 @@
 use crate::spec::{
-    RustcAbi, SanitizerSet, StackProbeType, Target, TargetMetadata, TargetOptions, base,
+    Cc, LinkerFlavor, Lld, RustcAbi, SanitizerSet, StackProbeType, Target, TargetMetadata,
+    TargetOptions, base,
 };
 
 // See https://developer.android.com/ndk/guides/abis.html#x86
@@ -7,6 +8,7 @@
 
 pub(crate) fn target() -> Target {
     let mut base = base::android::opts();
+    base.add_pre_link_args(LinkerFlavor::Gnu(Cc::Yes, Lld::No), &["--rtlib=compiler-rt"]);
 
     base.max_atomic_width = Some(64);
 
--- a/compiler/rustc_target/src/spec/targets/riscv64_linux_android.rs
+++ b/compiler/rustc_target/src/spec/targets/riscv64_linux_android.rs
@@ -1,10 +1,14 @@
 use std::borrow::Cow;
 
 use crate::spec::{
-    CodeModel, SanitizerSet, SplitDebuginfo, Target, TargetMetadata, TargetOptions, base,
+    Cc, CodeModel, LinkerFlavor, Lld, SanitizerSet, SplitDebuginfo, Target, TargetMetadata,
+    TargetOptions, base,
 };
 
 pub(crate) fn target() -> Target {
+    let mut base = base::android::opts();
+    base.add_pre_link_args(LinkerFlavor::Gnu(Cc::Yes, Lld::No), &["--rtlib=compiler-rt"]);
+
     Target {
         llvm_target: "riscv64-linux-android".into(),
         metadata: TargetMetadata {
@@ -24,7 +28,7 @@
             supported_sanitizers: SanitizerSet::ADDRESS,
             max_atomic_width: Some(64),
             supported_split_debuginfo: Cow::Borrowed(&[SplitDebuginfo::Off]),
-            ..base::android::opts()
+            ..base
         },
     }
 }
--- a/compiler/rustc_target/src/spec/targets/thumbv7neon_linux_androideabi.rs
+++ b/compiler/rustc_target/src/spec/targets/thumbv7neon_linux_androideabi.rs
@@ -11,6 +11,8 @@
 pub(crate) fn target() -> Target {
     let mut base = base::android::opts();
     base.add_pre_link_args(LinkerFlavor::Gnu(Cc::Yes, Lld::No), &["-march=armv7-a"]);
+    base.add_pre_link_args(LinkerFlavor::Gnu(Cc::Yes, Lld::No), &["--rtlib=compiler-rt"]);
+
     Target {
         llvm_target: "armv7-none-linux-android".into(),
         metadata: TargetMetadata {
--- a/compiler/rustc_target/src/spec/targets/x86_64_linux_android.rs
+++ b/compiler/rustc_target/src/spec/targets/x86_64_linux_android.rs
@@ -11,6 +11,7 @@
     base.features = "+mmx,+sse,+sse2,+sse3,+ssse3,+sse4.1,+sse4.2,+popcnt".into();
     base.max_atomic_width = Some(64);
     base.add_pre_link_args(LinkerFlavor::Gnu(Cc::Yes, Lld::No), &["-m64"]);
+    base.add_pre_link_args(LinkerFlavor::Gnu(Cc::Yes, Lld::No), &["--rtlib=compiler-rt"]);
     base.stack_probes = StackProbeType::Inline;
     base.supports_xray = true;
 
