diff -uNr a/src/runtime/src/libraries/System.IO.IsolatedStorage/src/System.IO.IsolatedStorage.csproj b/src/runtime/src/libraries/System.IO.IsolatedStorage/src/System.IO.IsolatedStorage.csproj
--- a/src/runtime/src/libraries/System.IO.IsolatedStorage/src/System.IO.IsolatedStorage.csproj	2026-04-23 00:22:06.585209338 +0800
+++ b/src/runtime/src/libraries/System.IO.IsolatedStorage/src/System.IO.IsolatedStorage.csproj	2026-04-30 05:56:11.766086829 +0800
@@ -25,12 +25,15 @@
              Link="Common\System\Obsoletions.cs" />
   </ItemGroup>
 
+  <!--
   <ItemGroup Condition="'$(TargetPlatformIdentifier)' == 'android' or '$(TargetPlatformIdentifier)' == 'ios' or '$(TargetPlatformIdentifier)' == 'maccatalyst' or '$(TargetPlatformIdentifier)' == 'tvos'">
     <Compile Include="System\IO\IsolatedStorage\Helper.AnyMobile.cs" />
     <Compile Include="System\IO\IsolatedStorage\IsolatedStorageFile.AnyMobile.cs" />
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetPlatformIdentifier)' != 'android' and '$(TargetPlatformIdentifier)' != 'ios' and '$(TargetPlatformIdentifier)' != 'maccatalyst' and '$(TargetPlatformIdentifier)' != 'tvos' and '$(TargetPlatformIdentifier)' != ''">
+  -->
+  <ItemGroup>
     <Compile Include="System\IO\IsolatedStorage\Helper.NonMobile.cs" />
     <Compile Include="System\IO\IsolatedStorage\IsolatedStorageFile.NonMobile.cs" />
   </ItemGroup>
