diff -uNr a/set.c b/set.c
--- a/set.c	2020-04-24 21:06:48.000000000 +0800
+++ b/set.c	2024-08-13 15:50:22.851001059 +0800
@@ -283,7 +283,8 @@
   switch (i->type) {
     case S_STRING:
       strcpy(i->u.sval, value_string);
-      if (i->action && !no_action) i->action(value_string);
+      //if (i->action && !no_action) i->action(value_string);
+	  if (i->action && !no_action) i->action(i->u.sval);
       break;
     case S_INTEGER:
       i->u.ival = atol(value_string);
diff -uNr a/set.h b/set.h
--- a/set.h	2020-04-24 21:06:48.000000000 +0800
+++ b/set.h	2024-08-13 15:50:35.523001059 +0800
@@ -51,7 +51,7 @@
 #ifdef __GNUC__
 #pragma GCC diagnostic ignored "-Wstrict-prototypes"
 #endif
-typedef void (*set_fn)();
+typedef void (*set_fn)(long);
 #ifdef __GNUC__
 #pragma GCC diagnostic error "-Wstrict-prototypes"
 #endif
