--- a/set.c
+++ b/set.c
@@ -283,11 +283,11 @@
   switch (i->type) {
     case S_STRING:
       strcpy(i->u.sval, value_string);
-      if (i->action && !no_action) i->action(value_string);
+      if (i->action && !no_action) i->action(const char * const value_string);
       break;
     case S_INTEGER:
       i->u.ival = atol(value_string);
-      if (i->action && !no_action) i->action(i->u.ival);
+      if (i->action && !no_action) i->action(long i->u.ival);
       break;
     case S_BOOL:
       i->u.bval = 0;
@@ -296,7 +296,7 @@
 	  i->u.bval = 1;
 	  break;
 	}
-      if (i->action && !no_action) i->action(i->u.bval);
+      if (i->action && !no_action) i->action(int i->u.bval);
       break;
     default:
       abort();
