diff -ur openbabel/src/plugin.cpp openbabel.mod/src/plugin.cpp
--- openbabel/src/plugin.cpp	2020-05-08 23:38:02.000000000 +0800
+++ openbabel.mod/src/plugin.cpp	2023-10-22 15:00:10.525918836 +0800
@@ -68,6 +68,7 @@
     if(DLHandler::openLib(*itr))
       count++;
   }
+  cout << "count = " << count << endl;
   if(!count) {
     string error = "No valid OpenBabel plugs found in "+TargetDir;
     obErrorLog.ThrowError(__FUNCTION__, error, obError);
