--- a/src/motiondetect.c
+++ b/src/motiondetect.c
@@ -108,8 +108,10 @@
   md->frameNum = 0;
 
   if(md->serializationMode != ASCII_SERIALIZATION_MODE && md->serializationMode != BINARY_SERIALIZATION_MODE) {
-    md->serializationMode = BINARY_SERIALIZATION_MODE;
+    vs_log_warn(md->conf.modName, "unknown file format %d, fallback to %d\n", md->serializationMode, ASCII_SERIALIZATION_MODE);
+    md->serializationMode = ASCII_SERIALIZATION_MODE;
   }
+  vs_log_info(md->conf.modName, "using file format %d\n", md->serializationMode);
 
   // TODO: get rid of shakiness parameter in the long run
   md->conf.shakiness = VS_MIN(10,VS_MAX(1,md->conf.shakiness));
